.github/workflows/lint.yml

```diff
@@ -38,6 +38,9 @@ jobs:
       - name: Install deps (with cache)
         run: pnpm install --child-concurrency 3
 
+      - name: Build
+        run: pnpm turbo --filter "@trpc/*" build
```

Comment:
Necessary because lint rules use type information that is more accurate after a build. Without a build before lint, we get odd lint rule failures [[example build](https://github.com/trpc/trpc/actions/runs/3778203385/jobs/6422483536)].

Reply:
Interesting - do you know why this happens? 🤔

Reply:
This can prob be configured in he turbo pipeline to get more optimized.

1. Setting build as a dependency for lint
2. Running just lint with turbo 

Might be able to split the linting up to get even faster CI if done correctly https://github.com/trpc/trpc/pull/3186